core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -349,8 +350,15 @@ final class ShuffleBlockFetcherIterator(
         results.put(new SuccessFetchResult(blockId, blockManager.blockManagerId,
           buf.size(), buf, false))
       } catch {
+        // If we see an exception, stop immediately.
+        case c: ClosedByInterruptException =>
+          // ClosedByInterruptException is an excepted exception when kill task,
+          // don't log the exception stack trace to avoid confusing users.
+          // See: SPARK-28340
+          logError(s"Interrupt occurred while fetching local blocks")
```

Comment:
Remove the interpolation. Should you name the exception here for clarity? In both cases you could log the exception message in the string here instead, to preserve some detail.

It'd be nice to avoid repeating the `results.put(...)` ; is this worth it?

```
case e: Exception =>
  e match {
     case ce: CloseByInterruptException => logError(...)
     case ex: Exception => logError(...)
  }
  results.put(...)
  return
```